### PR TITLE
Hotfix7

### PR DESCRIPTION
**Pull request by adityacp**
This PR includes fix for the following:-
1. Update answer paper marks as per latest attempts for non-coding questions.
2. Show last submitted answer during question revisits.
3. Prevent storing answers during skipping of questions.

### CODE REVIEW

File: yaksh/models.py [hunk 1]
```diff
@@ -1284,6 +1284,11 @@ def get_question_answers(self):
                 }]
         return q_a
 
+    def get_latest_answer(self,question_id):
+        all_answers = self.answers.filter(question=question_id)
+        if all_answers:
```
Comment (prabhuramachandran) [inline yaksh/models.py]: This looks wrong, as I am not sure that all_answers[0] is indeed the latest answer, instead use `self.answers.latest('id')`?

File: yaksh/test_views.py [hunk 2]
```diff
@@ -3378,3 +3379,387 @@ def test_show_statistics_for_student_per_attempt(self):
                                 [1, 1])
         self.assertEqual(response.context['attempts'][0], 1)
         self.assertEqual(response.context['total'], 1)
+
+
+class TestCheck(TestCase):
```
Comment (prabhuramachandran) [inline yaksh/test_views.py]: This class name is not nice, can you pick TestQuestionPaper or so?

File: yaksh/views.py [hunk 3]
```diff
@@ -594,11 +597,17 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None):
         else:
             user_answer = request.POST.get('answer')
         if not user_answer:
-            msg = ["Please submit a valid option or code"]
+            msg = "Please submit a valid option or code"
```
Comment (prabhuramachandran) [inline yaksh/views.py]: Why is this option or code?  Isn't it just an option?

Comment (prabhuramachandran) [reply yaksh/views.py]: Or maybe option/value?

Comment (adityacp) [reply yaksh/views.py]: This message is displayed for both code and non-code questions if user answer is empty. So the message is common for both.

File: yaksh/models.py [hunk 4]
```diff
@@ -1284,6 +1284,9 @@ def get_question_answers(self):
                 }]
         return q_a
 
+    def get_latest_answer(self, question_id):
+        return self.answers.filter(question=question_id).order_by("-id").last()
```
Comment (prabhuramachandran) [inline yaksh/models.py]: This looks wrong, it should be `self.answers.filter(question=question_id).order_by("id").last()`

**Comment by prabhuramachandran** [comment]
My only issue is that the tests did not catch the last answer issue that was seen.  Please add a test for this ASAP.  I may merge this but please add this test.

**Pull request merged**